breathe/directives/function.py

```diff
@@ -133,21 +133,24 @@ def _parse_args(self, function_description: str) -> Optional[cpp.ASTParametersQu
                                       location=self.get_source_info(),
                                       config=self.config)
         paramQual = parser._parse_parameters_and_qualifiers(paramMode='function')
-        # now erase the parameter names
-        for p in paramQual.args:
-            if p.arg is None:
-                assert p.ellipsis
-                continue
-            declarator = p.arg.type.decl
-            while hasattr(declarator, 'next'):
-                if isinstance(declarator, cpp.ASTDeclaratorParen):
-                    assert hasattr(declarator, 'inner')
-                    declarator = declarator.inner  # type: ignore
-                else:
-                    declarator = declarator.next  # type: ignore
-            assert hasattr(declarator, 'declId')
-            declarator.declId = None  # type: ignore
-            p.arg.init = None  # type: ignore
+        # strip everything that doesn't contribute to overloading
+
+        def stripParamQual(paramQual):
+            for p in paramQual.args:
+                if p.arg is None:
+                    assert p.ellipsis
+                    continue
+                declarator = p.arg.type.decl
+                while hasattr(declarator, 'next'):
+                    if isinstance(declarator, cpp.ASTDeclaratorParen):
+                        assert hasattr(declarator, 'inner')
+                        declarator = declarator.inner  # type: ignore
+                    else:
+                        declarator = declarator.next  # type: ignore
+                assert hasattr(declarator, 'declId')
+                declarator.declId = None  # type: ignore
+                p.arg.init = None  # type: ignore
+        stripParamQual(paramQual)
```

Comment:
What is the advantage of moving this into a function that is immediately executed? I'm a bit out of touch with Python.

Reply:
Ah, you end up calling it recursively.